File: services/file-processing-pipeline/Domain/Entities/FileMetadata.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace FileProcessingPipeline.Domain.Entities;

public class FileMetadata
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; } = ObjectId.GenerateNewId().ToString();
    
    [BsonElement("fileId")]
    [BsonRepresentation(BsonType.String)]
    public Guid FileId { get; set; }
    
    [BsonElement("userId")]
    [BsonRepresentation(BsonType.String)]
    public Guid UserId { get; set; }
    
    [BsonElement("originalFileName")]
    public string OriginalFileName { get; set; } = string.Empty;
    
    [BsonElement("storedFileName")]
    public string StoredFileName { get; set; } = string.Empty;
    
    [BsonElement("contentType")]
    public string ContentType { get; set; } = string.Empty;
    
    [BsonElement("fileSize")]
    public long FileSize { get; set; }
    
    [BsonElement("hash")]
    public string Hash { get; set; } = string.Empty;
    
    [BsonElement("hashAlgorithm")]
    public string HashAlgorithm { get; set; } = "SHA256";
    
    [BsonElement("minioObjectKey")]
    public string MinioObjectKey { get; set; } = string.Empty;
    
    [BsonElement("minioOriginalObjectKey")]
    public string MinioOriginalObjectKey { get; set; } = string.Empty;
    
    [BsonElement("minioBucket")]
    public string MinioBucket { get; set; } = string.Empty;
    
    [BsonElement("originalBucket")]
    public string OriginalBucket { get; set; } = string.Empty;
    
    [BsonElement("isEncrypted")]
    public bool IsEncrypted { get; set; }
    
    [BsonElement("isDecryptionValidated")]
    public bool IsDecryptionValidated { get; set; }
    
    [BsonElement("description")]
    public string? Description { get; set; }
    
    [BsonElement("status")]
    public FileProcessingStatus Status { get; set; } = FileProcessingStatus.Pending;
    
    [BsonElement("errorMessage")]
    public string? ErrorMessage { get; set; }
    
    [BsonElement("createdAt")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    [BsonElement("processedAt")]
    public DateTime? ProcessedAt { get; set; }
    
    [BsonElement("customMetadata")]
    public Dictionary<string, object> CustomMetadata { get; set; } = new();
}

public enum FileProcessingStatus
{
    Pending,
    Processing,
    Completed,
    Failed
}

File: services/spring-visualizer/src/main/java/com/fileshare/visualizer/model/FileMetadata.java
package com.fileshare.visualizer.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Document(collection = "file_metadata")
public class FileMetadata {
    @Id
    private String id;

    @Field("fileId")
    private UUID fileId;

    @Field("userId")
    private UUID userId;

    @Field("originalFileName")
    private String originalFileName;

    @Field("contentType")
    private String contentType;

    @Field("fileSize")
    private Long fileSize;

    @Field("hash")
    private String hash;

    @Field("isEncrypted")
    private Boolean isEncrypted;

    @Field("description")
    private String description;

    @Field("status")
    private String status;

    @Field("createdAt")
    private LocalDateTime createdAt;

    @Field("processedAt")
    private LocalDateTime processedAt;

    @Field("minioObjectKey")
    private String minioObjectKey;
}
